apps/src/templates/sectionProgressV2/MoreDetailsDialog.jsx

```diff
@@ -1,27 +1,118 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import i18n from '@cdo/locale';
-import Button from '@cdo/apps/templates/Button';
 import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
-import {Heading3, Heading5} from '@cdo/apps/componentLibrary/typography';
+import {
+  Heading3,
+  Heading6,
+  StrongText,
+  BodyThreeText,
+} from '@cdo/apps/componentLibrary/typography';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import {ITEM_TYPE} from './ItemType';
+import ProgressIcon from './ProgressIcon';
+import styles from './progress-key-popup.module.scss';
 
 export default function MoreDetailsDialog({hasValidation, onClose}) {
+  const validatedLevelsContent = () => (
+    <div className={styles.item}>
+      <ProgressIcon itemType={ITEM_TYPE.VALIDATED} />
+      <BodyThreeText>
+        <StrongText>{i18n.validated() + ': '}</StrongText>
+        {i18n.progressLegendDetailsValidated()}
+      </BodyThreeText>
+    </div>
+  );
   return (
     <AccessibleDialog onClose={onClose}>
-      <div role="region">
-        <div>
-          <Heading3>{i18n.progressTrackingIconKey()}</Heading3>
+      <div role="region" className={styles.dialog}>
+        <Heading3>{i18n.progressTrackingIconKey()}</Heading3>
+        <button type="button" onClick={onClose} className={styles.xCloseButton}>
+          <i id="x-close" className="fa-solid fa-xmark" />
+        </button>
+        <hr />
+        <Heading6>{i18n.assignmentCompletionStates()}</Heading6>
+        <div className={styles.item}>
+          <ProgressIcon itemType={ITEM_TYPE.NOT_STARTED} />
+          <BodyThreeText>
+            <StrongText>{i18n.notStarted() + ': '}</StrongText>
+            {i18n.progressLegendDetailsNotStarted()}
+          </BodyThreeText>
         </div>
-        <div>
-          <Heading5>Info info info</Heading5>
+        <div className={styles.item}>
+          <ProgressIcon itemType={ITEM_TYPE.NO_ONLINE_WORK} />
+          <BodyThreeText>
+            <StrongText>{i18n.noOnlineWork() + ': '}</StrongText>
+            {i18n.progressLegendDetailsNoOnlineWork()}
+          </BodyThreeText>
+        </div>
+        <div className={styles.item}>
+          <ProgressIcon itemType={ITEM_TYPE.IN_PROGRESS} />
+          <BodyThreeText>
+            <StrongText>{i18n.inProgress() + ': '}</StrongText>
+            {i18n.progressLegendDetailsInProgress()}
+          </BodyThreeText>
+        </div>
+        <div className={styles.item}>
+          <ProgressIcon itemType={ITEM_TYPE.SUBMITTED} />
+          <div>
```

Comment:
Again, I tried to get rid of all the `divs` I could here - initially I had the `SafeMarkdown` inside the `BodyThreeText` - but that threw a warning.  In order to sue the grid system I needed to wrap all of this into something, thus using the `div` here.